test/syntax.jl

```diff
@@ -3747,3 +3747,16 @@ b54805 = 2
 end
 using .Export54805
 @test b54805 == 2
+
+# Test that lowering doesn't accidentally put a `Module` in the Method name slot
+let src = @Meta.lower let capture=1
+    global foo_lower_block
```

Comment:
IIUC, the reason this must be in lowering (and wrong to inspect the signature unless lowering instructs that it must) is that this line could be:
```suggestion
    global const foo_lower_block = Base.redirect_stdin
```
And looking at the signature would wrongly name this Method as `Base.RedirectStdStream(0)`, when syntactically it has the different intended name `foo_lower_block`